src/UI/Implementation/Component/Input/Field/Renderer.php

```diff
@@ -266,10 +269,40 @@ protected function renderInputField(Template $tpl, Input $input, $id) {
 			$tpl->setVariable("ID", $id);
 			$tpl->parseCurrentBlock();
 		}
+		//specific options for textarea component.
+		if($input instanceof TextArea){
+			$tpl = $this->renderTextareaField($tpl, $input);
+		}
 
 		return $tpl->get();
 	}
 
+	protected function renderTextareaField(Template $tpl, TextArea $input)
+	{
+		global $DIC;
+
+		$lng = $DIC->language();
+		$lng->loadLanguageModule("form");
```

Comment:
I have to deal with language variables in different places and some of them should be used in the JS. IMO this is something that we will need in other scenarios. @Amstutz  @klees Are you OK with loading the language module in the Form container?
/src/UI/Implementation/Component/Input/Container/Form/Form.php

public function __construct(array $inputs) {
        global $DIC;
        $this->lng = $DIC->language();
        $this->lng->loadLanguageModule("form");
        ...
    ...

Reply:
Hi @xus. you should not use the $DIC in the renderer on the field. The DI here works hierarchically, meaning we inject the dependencies directly on the abstractComponentRenderer. Meaning you have access to the correct language dependency through $this->lng (inside the renderer). The component should not be aware of any language. 

You have a valid point with the language module. Obviously, there is some central place, needing to load the language module for the form. The renderer of the form would probably be the right place to do that. Maybe with a nice if before to check if already loaded. @klees do you agree?

Reply:
In general I'm fine with @Amstutz's solution. But I still would like to understand the general problem a bit more. @xus: What are these lang vars you want to use? How are they form specific and not e.g. part of "common"?

Reply:
@Amstutz AFAIK ilLanguage already checks itself if a lang module is already loaded.

Reply:
Thank you guys for your attention on this topic.
The UI textarea needs multilanguage in two places when number of characters is limited by the user:
- Using the byline to show the "Minimum:" and "Maximum" number of characters allowed.
- Displaying the character counter via JS. (in trunk -> form#:#form_chars_remaining#:#Characters remaining:)
	$lng->toJS("form_chars_remaining");

I thought that since we are using forms we should use the form module. Probably in the future other Inputs will need multilanguage and identify somehow all this lang vars can be useful.
Obviously I will be completely fine with your final decision/suggestions.

Reply:
Thx for the info @xus. I think we might need to have a look into the organization of the lang vars (for UI framework at least?) at some point in the future, but atm I'm fine with loading the "form"-module in the according renderer, as @Amstutz suggests.

Reply:
Hi @Amstutz, if I understood your first comment properly, you said that we can't use $DIC in the input renderer. I see that we can use directly $this->lng but I have to load a language module somehow and I don't see the way you want us to do this.(I have only access to "common" vars).

My first idea was add a new method in the AbstractComponentRenderer to do this task, something like:
final public function loadLangModule($lang)
{
	$this->lng->loadLanguageModule($lang);
}

allowing to load the module from the renderer:
protected function renderTextareaField(Template $tpl, TextArea $input)
{
	$this->loadLangModule('form');
	...
	...

But looking other Pull Requests I see that other developers are using DIC inside the specific renderers. Look at the file:
src/UI/Implementation/Component/Input/Field/Renderer.php

in PRS:
https://github.com/ILIAS-eLearning/ILIAS/pull/1142/files
https://github.com/ILIAS-eLearning/ILIAS/pull/1136/files

Reply:
Hi @xus, thanks for pointing this out. I did not perform a review on the first PR yet, so I can not yet say anything about that. I will look into that when I perform the review. For the second, I did not notice. I now commented on it, thx.

Do you understand why it is important for us, that $DIC is not used through global access? Do you agree that this is important?

You are right in pointing out, that there is currently now way to load the module, since the lng in AbstractComponentRenderer is kept private. I would be fine with your suggestion of opening up the interface there slightly by allowing:
```
final public function loadLangModule($lang)
{
  $this->lng->loadLanguageModule($lang);
}
```

Reply:
I would propose to just introduce a "ui"-module and load it anyway. This also would also organize the variables nicely. Opening that interface IMO is to much. Maybe @matthiaskunkel has an opinion?

Reply:
I am also fine with the ui-module, as long as @xus has some option to finish up this job ;-).

Reply:
Thank you guys for the feedback.

> Do you understand why it is important for us, that $DIC is not used through global access? Do you agree that this is important?

Yes, I think so. I agree with you guys that injecting dependencies and avoid global vars is the way to go for multiple and safer reasons.

I will add the function in to AbstractComponentRenderer, thanks @Amstutz 

I also think that ui-module can be useful, thanks @klees

Reply:
*I will add the function in to AbstractComponentRenderer, thanks @Amstutz*

Please don't. I don't think people should be able or required to load lang modules while rendering.

Just put your lang-var in an "ui"-module and load it [here](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/src/UI/Implementation/Render/DefaultRendererFactory.php#L44).

Thx for your patience.